lale/lib/snapml/decision_tree_regressor.py

```diff
@@ -0,0 +1,293 @@
+        X = lale.datasets.data_schemas.strip_schema(X)
+        if predict_params is None:
+            return self._wrapped_model.predict(X)
+        else:
+            return self._wrapped_model.predict(X, **predict_params)
+
+
+_hyperparams_schema = {
+    "description": "Hyperparameter schema.",
+    "allOf": [
+        {
+            "description": "This first sub-object lists all constructor arguments with their types, one at a time, omitting cross-argument constraints.",
+            "type": "object",
+            "relevantToOptimizer": ["max_depth", "max_features", "hist_nbins"],
+            "additionalProperties": False,
+            "properties": {
+                "criterion": {
+                    "enum": ["mse"],
+                    "default": "mse",
+                    "description": "Function to measure the quality of a split.",
+                },
+                "splitter": {
+                    "enum": ["best"],
+                    "default": "best",
+                    "description": "The strategy used to choose the split at each node.",
+                },
+                "max_depth": {
+                    "anyOf": [
+                        {
+                            "type": "integer",
+                            "minimum": 1,
+                            "minimumForOptimizer": 3,
+                            "maximumForOptimizer": 5,
+                        },
+                        {
+                            "enum": [None],
+                            "description": "Nodes are expanded until all leaves are pure or until all leaves contain less than min_samples_leaf samples.",
+                        },
+                    ],
+                    "default": None,
+                    "description": "The maximum depth of the tree.",
+                },
+                "min_samples_leaf": {
+                    "anyOf": [
+                        {
+                            "type": "integer",
+                            "minimum": 1,
+                            "forOptimizer": False,
+                            "description": "Consider min_samples_leaf as the minimum number.",
+                        },
+                        {
+                            "type": "number",
+                            "minimum": 0.0,
+                            "exclusiveMinimum": True,
+                            "maximum": 0.5,
+                            "description": "min_samples_leaf is a fraction and ceil(min_samples_leaf * n_samples) are the minimum number of samples for each node.",
+                        },
+                    ],
+                    "default": 1,
+                    "description": "The minimum number of samples required to be at a leaf node.",
+                },
+                "max_features": {
+                    "anyOf": [
+                        {
+                            "type": "integer",
+                            "minimum": 1,
```

Comment:
Should add a laleMax here for checking against the number of columns in the dataset, like this: https://github.com/IBM/lale/blob/5013699e88add4e79000c2f889c2c8450e7795e0/lale/lib/sklearn/decision_tree_regressor.py#L142

Reply:
Done